src/components/WalletDropdown/TransactionMenu.tsx

```diff
@@ -0,0 +1,159 @@
+import { Trans } from '@lingui/macro'
+import { useWeb3React } from '@web3-react/core'
+import { getYear, isSameDay, isSameWeek, isSameYear } from 'date-fns'
+import { useCallback, useMemo } from 'react'
+import { useAppDispatch } from 'state/hooks'
+import styled from 'styled-components/macro'
+
+import { useAllTransactions } from '../../state/transactions/hooks'
+import { clearAllTransactions } from '../../state/transactions/reducer'
+import { TransactionDetails } from '../../state/transactions/types'
+import { TransactionSummary } from '../AccountDetailsV2'
+import { SlideOutMenu } from './SlideOutMenu'
+
+const Divider = styled.div`
+  margin-top: 16px;
+  border-bottom: ${({ theme }) => `1px solid ${theme.backgroundOutline}`};
+`
+
+const TransactionListWrapper = styled.div`
+  ${({ theme }) => theme.flexColumnNoWrap};
+`
+
+interface TransactionInformation {
+  title: string
+  transactions: TransactionDetails[]
+}
+
+const TransactionTitle = styled.span`
+  padding-bottom: 8px;
+  padding-top: 20px;
+  padding-left: 12px;
+  padding-right: 12px;
+  font-weight: 600;
+  color: ${({ theme }) => theme.textTertiary};
+`
+
+const renderTransactions = (transactionInformation: TransactionInformation) => {
+  const { title, transactions } = transactionInformation
+
+  return (
+    <TransactionListWrapper key={title}>
+      <TransactionTitle>{title}</TransactionTitle>
+      {transactions.map((transactionDetails, i) => (
+        <TransactionSummary key={i} transactionDetails={transactionDetails} />
+      ))}
+    </TransactionListWrapper>
+  )
+}
+
+const getConfirmedTransactions = (confirmedTransactions: Array<TransactionDetails>) => {
+  const ONE_DAY = 1000 * 60 * 60 * 24
+  const THIRTY_DAYS = ONE_DAY * 30
+  const NOW = new Date().getTime()
+
+  const TODAY: Array<TransactionDetails> = []
+  const CURRENT_WEEK: Array<TransactionDetails> = []
+  const LAST_30_DAYS: Array<TransactionDetails> = []
+  const CURRENY_YEAR: Array<TransactionDetails> = []
+  const yearMap: { [key: string]: Array<TransactionDetails> } = {}
```

Comment:
The logic is as follows, found in notion here: https://www.notion.so/uniswaplabs/Transaction-State-History-Web-0ea7b872504148cfaab557088bcd8314


1. **Pending section:**  any tx that were submitted by this wallet and are currently pending.
2. **Today:**  any tx that were completed (success or fail) today
3. **This week:**  all tx that were completed (successfully) this week
4. **Past 30 days:**  any tx that were completed in the last 30 days
5. **Current year (2022):**  all tx that were completed in the current calendar year
6. **Subsequent past year sections (2021, 2020):**  group tx by the calendar year in which they were completed. This is important